homeassistant/components/homeassistant/scene.py

```diff
@@ -271,6 +282,28 @@ async def create_service(call: ServiceCall) -> None:
         SCENE_DOMAIN, SERVICE_CREATE, create_service, CREATE_SCENE_SCHEMA
     )
 
+    async def delete_service(call: ServiceCall) -> None:
+        """Delete a dynamically created scene."""
+        entity_id = call.data[CONF_ENTITY_ID]
+        scene = platform.entities.get(entity_id)
+        if scene is None:
+            _LOGGER.error("The scene %s does not exist", entity_id)
+            return
+        if not isinstance(scene, HomeAssistantScene):
+            _LOGGER.error("Entity %s is not a scene", entity_id)
+            return
+        if not scene.from_service:
+            _LOGGER.error(
```

Comment:
Raise `ServiceValidationError` for bad user input and `HomeAssistantError` for other problems. Pass an argument with the message to the exception. The message can have translations and will show in the frontend.

https://developers.home-assistant.io/docs/core/platform/raising_exceptions/